org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/JDTLanguageServer.java

```diff
@@ -411,8 +411,8 @@ public void didChangeConfiguration(DidChangeConfigurationParams params) {
 			Collection<IPath> rootPaths = preferenceManager.getPreferences().getRootPaths();
 			@SuppressWarnings("unchecked")
 			Preferences prefs = Preferences.createFrom((Map<String, Object>) settings);
+			prefs.setRootPaths(rootPaths);
```

Comment:
`RootPath` is set before updating preferences, to ensure the listener will get the latest root paths.